### DISCUSSION TITLE
UsersResource.search([attribute], x, x) Doesn't work

### DISCUSSION THREAD

**Discussion by jaggaer-npasic** [keycloak/keycloak#8446] [Q&A]
Hi,
Can someone help me with this issue?
According to documentation (https://www.keycloak.org/docs-api/11.0/javadocs/org/keycloak/admin/client/resource/UsersResource.html#search-java.lang.String-java.lang.Integer-java.lang.Integer-) 

_Search for users whose username or email matches the value provided by search. The search argument also allows finding users by specific attributes as follows:
id: - Find users by identifier. For instance, id:aa497859-bbf5-44ac-bf1a-74dbffcaf197_

This should allow us to get the user by email, but this doesn't work.
_search - the value to search. It can be the username, email or any of the supported options to query based on user attributes_

I tried to call this method usersResource.search("email:user@email.com", 0,100); but in case when any other attribute than id is used it doesn't retrieve anything, only id works.

Am I missing something?

**Comment by stianst** [comment]
Take a look at https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_users_resource

The search query parameter does a fuzzy search across many attributes (?search=user@emai.com), while if you want to explicitly look for usrs by email use the email query parameter (?email=user@email.com).

**Reply by jaggaer-npasic** [reply]
Hi, Thanks for your reply.
This search does a search on  _**/auth/admin/realms**_ which is an **admin** resource, and the admin resource is not searchable by using **keycloak-admin-client** which uses _**/auth/realms/**_ .
Here, I'm talking about default **keycloak-admin-client** usage.
I'm not willing to create a new REST service in to call **admin** resource.

**Reply by stianst** [reply]
I'm not quite following, as the keycloak-admin-client is just a Java client to the Admin APIs that I linked to. If you look at:

https://github.com/keycloak/keycloak/blob/master/integration/admin-client/src/main/java/org/keycloak/admin/client/resource/UsersResource.java#L76

You can do what I suggested there, just pass null for the fields you don't care about.

**Reply by jaggaer-npasic** [reply]
I use at the moment v 11.0.2 of Keycloak.
I did like you suggested, but it doesn't work. null or "" doesn't give the desired result. In both cases, it returns a list, instead of an exact match.
Example:
you have users with emails:
abc@domain.com
bc@doman.com
c@domain.com

and if you send a request searching for user with an email bc@domain.com you'll get as a result two objects
abc@domain.com and
bc@doman.com

While https://github.com/keycloak/keycloak/blob/master/integration/admin-client/src/main/java/org/keycloak/admin/client/resource/UsersResource.java#L131 works only for id: not for other attributes, like email..

**Reply by stianst** [reply]
I'd suggest trying with the latest release, and if it's still not working open a bug

**Reply by jaggaer-npasic** [reply]
Ok. Will do. Thanks
